src/main/deltachat/webxdc.ts

```diff
@@ -248,13 +248,80 @@ export default class DCWebxdc extends SplitOut {
         internet_access: webxdcInfo['internetAccess'],
       }
 
+      type setPermissionRequestHandler =
+        typeof webxdcWindow.webContents.session.setPermissionRequestHandler
+      type permission_arg = Parameters<
+        Exclude<Parameters<setPermissionRequestHandler>[0], null>
+      >[1]
+      // TODO some (poorly written?) apps might require a refresh
+      // after a permission has been granted,
+      // but we don't support it because
+      // 1. There is no way to refresh a webxdc
+      // 2. There is the WebRTC 500 exhaustion hack which doesn't allow the app
+      //     to load. We should wait there instead of immediately closing.
+      const grantedPermissions = new Set<permission_arg>([
+        // because games might lock the pointer
+        'pointerLock',
+        // games might do that too
+        'fullscreen',
+      ])
+      const userControllablePermissions: permission_arg[] = [
+        // TODO should add more permissions, I added just a few for a POC.
+        //
+        // Or, we should instead dynamically add items here as the app
+        // makes permission requests (see `permission_handler`).
+        // Would we have to `window.setMenu()` each time?
+        // Can we utilize `visible: boolean` attribute instead?
+        'media',
+        // TODO `navigator.geolocation.getCurrentPosition(console.log, console.error)`
+        // fails because
+        // `Network location provider at 'https://www.googleapis.com/' : ERR_NAME_NOT_RESOLVED.`
+        'geolocation',
+        'notifications',
+        'display-capture',
```

Comment:
I'd remove `notifications` and `display-capture` for now:
- `notifications` -> webxdc will have it's own api for notifications and I think this notification api want network access
-  `display-capture` - probably does not work without making a custom ui for choosing what to share?